### PR TITLE
remote: ensure that client connection is not established twice

### PR DESCRIPTION
**Pull request by tonistiigi**
Because remote driver implements Info() by calling Client() internally, two instances on Client are created backed by separate TCP connection. This hack avoids it and improves performance.

### CODE REVIEW

File: driver/remote/driver.go [hunk 1]
```diff
@@ -78,12 +85,18 @@ func (d *Driver) Rm(ctx context.Context, force, rmVolume, rmDaemon bool) error {
...
-			return d.Dial(ctx)
-		}),
-	}, opts...)
-	return client.New(ctx, "", opts...)
+	d.clientOnce.Do(func() {
+		opts = append([]client.ClientOpt{
+			client.WithContextDialer(func(ctx context.Context, _ string) (net.Conn, error) {
+				return d.Dial(ctx)
+			}),
+			client.WithTracerDelegate(delegated.DefaultExporter),
```
Comment (tonistiigi) [inline driver/remote/driver.go:93]: This is pretty 😬 , normally called by `DriverHandle`. Mostly weird as other drivers don't need to call it.

Comment (crazy-max) [reply driver/remote/driver.go:93]: But do not `client.WithTracerDelegate` already set as opt in https://github.com/docker/buildx/blob/cfef22ddf0ccff52a8aced77c6115249d78b914b/driver/remote/driver.go#L80 via https://github.com/docker/buildx/blob/cfef22ddf0ccff52a8aced77c6115249d78b914b/driver/manager.go#L167

Comment (tonistiigi) [reply driver/remote/driver.go:93]: This is how it works in regular mode and for other drivers. But for remote driver the call path can be that `Info()` is called that internally calls `Client()` (and that client is now cached). In that case no delegate is set up.

Maybe a possible solution would be to move `ClientOpt` as part of `InitConfig`

Comment (crazy-max) [reply driver/remote/driver.go:93]: > Maybe a possible solution would be to move `ClientOpt` as part of `InitConfig`

Ah right yeah that would be neat

Comment (tonistiigi) [reply driver/remote/driver.go:93]: This still would mean that every driver would need to handle the `ClientOpt` and if they don't then at least the tracing would not work.

**Review by crazy-max — APPROVED** [review]

**Pull request merged**